clients/src/main/java/org/apache/kafka/clients/admin/internals/AdminMetadataManager.java

```diff
@@ -280,6 +281,9 @@ public void updateFailed(Throwable exception) {
         if (exception instanceof AuthenticationException) {
             log.warn("Metadata update failed due to authentication error", exception);
             this.fatalException = (ApiException) exception;
+        } else if (exception instanceof ClusterAuthorizationException) {
```

Comment:
should this be `AuthorizationException` here since all authZ exceptions are non-retriable?